File: Code/@Region/Region.m
%% Class for source, gate and drain.
classdef Region < handle
    properties (GetAccess = public, SetAccess = {?Transistor, ?Region})
        phi
        carrier_density
        DOS
        range_start
        range_end
    end
    
    methods
        function obj = Region(range_start, range_end)
            % phi, carrier density, DOS are set separately
            % (those aren't constant)
            obj.range_start = range_start;
            obj.range_end = range_end;
        end
        
        function rg = range(obj)
            rg = obj.range_start:obj.range_end;
        end
    end
end

File: Code/region.py
import numpy as np

# Class for source, gate and drain.
class Region:
    phi: np.ndarray
    carrier_density: np.ndarray
    DOS: np.ndarray

    def __init__(self, range_start: int, range_end: int) -> None:
        self.range_start = range_start
        self.range_end = range_end

    # phi
    # carrier_density
    # DOS
    # range_start
    # range_end

    def slice(self) -> slice:
        return slice(self.range_start, self.range_end)
